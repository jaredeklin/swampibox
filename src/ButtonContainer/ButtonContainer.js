import React from 'react';
import Button from '../Button/Button';
import './buttonContainer.css';
import PropTypes from 'prop-types';

export const ButtonContainer = ({getData, active, favorites}) => {
  const category = ['People', 'Planets', 'Vehicles', 'Favorites'];

  const createButtons = category.map((type, index) => {

    let activeToggle = 'category-button';

    if (type === active[0]) {
      activeToggle = 'category-button active';
    }
    return (
      <Button category={type} 
        getData={getData} 
        className={activeToggle} 
        favorites={favorites}
        key={index}/>);
  });

  return (
    <div className='button-container'>
      {createButtons}
    </div>
  );
};

ButtonContainer.propTypes = {
  getData: PropTypes.func,
  active: PropTypes.array,
  favorites: PropTypes.number
};