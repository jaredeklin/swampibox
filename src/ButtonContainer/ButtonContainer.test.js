import React from 'react';
import { shallow, mount } from 'enzyme';
import { ButtonContainer } from './ButtonContainer';

describe('ButtonContainer', () => {

  let mockActive = [];

  it('should match the snapshot', () => {
    let wrapper = shallow(<ButtonContainer active={mockActive} />)
    
    expect(wrapper).toMatchSnapshot()
  })

  it('should add an active class if in the active array', () => {
    mockActive = ['People'];
    let wrapper = shallow(<ButtonContainer active={mockActive} />)

    expect(wrapper).toMatchSnapshot()
  })

})