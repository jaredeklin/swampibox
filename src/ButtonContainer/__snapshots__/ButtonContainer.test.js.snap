// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should add an active class if in the active array 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    active={
        Array [
            "People",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          category="People"
          className="category-button active"
          favorites={undefined}
          getData={undefined}
/>,
        <Unknown
          category="Planets"
          className="category-button"
          favorites={undefined}
          getData={undefined}
/>,
        <Unknown
          category="Vehicles"
          className="category-button"
          favorites={undefined}
          getData={undefined}
/>,
        <Unknown
          category="Favorites"
          className="category-button"
          favorites={undefined}
          getData={undefined}
/>,
      ],
      "className": "button-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "category": "People",
          "className": "category-button active",
          "favorites": undefined,
          "getData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "category": "Planets",
          "className": "category-button",
          "favorites": undefined,
          "getData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "category": "Vehicles",
          "className": "category-button",
          "favorites": undefined,
          "getData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "category": "Favorites",
          "className": "category-button",
          "favorites": undefined,
          "getData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            category="People"
            className="category-button active"
            favorites={undefined}
            getData={undefined}
/>,
          <Unknown
            category="Planets"
            className="category-button"
            favorites={undefined}
            getData={undefined}
/>,
          <Unknown
            category="Vehicles"
            className="category-button"
            favorites={undefined}
            getData={undefined}
/>,
          <Unknown
            category="Favorites"
            className="category-button"
            favorites={undefined}
            getData={undefined}
/>,
        ],
        "className": "button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "category": "People",
            "className": "category-button active",
            "favorites": undefined,
            "getData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "category": "Planets",
            "className": "category-button",
            "favorites": undefined,
            "getData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "category": "Vehicles",
            "className": "category-button",
            "favorites": undefined,
            "getData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "category": "Favorites",
            "className": "category-button",
            "favorites": undefined,
            "getData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ButtonContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    active={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          category="People"
          className="category-button"
          favorites={undefined}
          getData={undefined}
/>,
        <Unknown
          category="Planets"
          className="category-button"
          favorites={undefined}
          getData={undefined}
/>,
        <Unknown
          category="Vehicles"
          className="category-button"
          favorites={undefined}
          getData={undefined}
/>,
        <Unknown
          category="Favorites"
          className="category-button"
          favorites={undefined}
          getData={undefined}
/>,
      ],
      "className": "button-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "category": "People",
          "className": "category-button",
          "favorites": undefined,
          "getData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "category": "Planets",
          "className": "category-button",
          "favorites": undefined,
          "getData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "category": "Vehicles",
          "className": "category-button",
          "favorites": undefined,
          "getData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "category": "Favorites",
          "className": "category-button",
          "favorites": undefined,
          "getData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            category="People"
            className="category-button"
            favorites={undefined}
            getData={undefined}
/>,
          <Unknown
            category="Planets"
            className="category-button"
            favorites={undefined}
            getData={undefined}
/>,
          <Unknown
            category="Vehicles"
            className="category-button"
            favorites={undefined}
            getData={undefined}
/>,
          <Unknown
            category="Favorites"
            className="category-button"
            favorites={undefined}
            getData={undefined}
/>,
        ],
        "className": "button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "category": "People",
            "className": "category-button",
            "favorites": undefined,
            "getData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "category": "Planets",
            "className": "category-button",
            "favorites": undefined,
            "getData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "category": "Vehicles",
            "className": "category-button",
            "favorites": undefined,
            "getData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "category": "Favorites",
            "className": "category-button",
            "favorites": undefined,
            "getData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
